packages/reports/src/Components/Form/CompositeCalendar/two-month-picker.tsx

```diff
@@ -1,63 +1,86 @@
     };
 
     /**
-     * Only allow previous months to be available to navigate. Disable other periods
+     * Navigate to date
      *
-     * @param date
-     * @param range
-     * @returns {boolean}
+     * @param {moment.Moment} date
+     * @returns {void}
      */
-    const validateFromArrows = (date, range) => {
-        return range === 'year' || diffInMonths(epochToMoment(left_pane_date), date) !== -1;
+    const navigateTo = (date: moment.Moment): void => {
+        setLeftPaneDate(subMonths(date.toISOString(), 1).unix());
+        setRightPaneDate(toMoment(date).unix());
     };
 
     /**
-     * Validate values to be date_from < date_to
+     * Only allow previous months to be available to navigate. Disable other periods
+     *
+     * @param {moment.Moment} date
+     * @param {Extract<moment.DurationInputArg2, 'month'>} range
+     * @returns {boolean}
      */
-    const shouldDisableDate = date => {
-        return isPeriodDisabled(date.unix());
+    const validateFromArrows = (date: moment.Moment, range: Extract<moment.DurationInputArg2, 'month'>): boolean => {
+        // return range === 'year' || diffInMonths(epochToMoment(left_pane_date), date) !== -1;
+        return diffInMonths(epochToMoment(left_pane_date), date) !== -1;
     };
 
     /**
      * Only allow next month to be available to navigate (unless next month is in the future).
      * Disable other periods
      *
-     * @param date
-     * @param range
+     * @param {moment.Moment} date
+     * @param {Extract<moment.DurationInputArg2, 'month'>} range
      * @returns {boolean}
      */
-    const validateToArrows = (date, range) => {
-        if (range === 'year') return true; // disallow year arrows
+    const validateToArrows = (date: moment.Moment, range: Extract<moment.DurationInputArg2, 'month'>): boolean => {
+        // if (range === 'year') return true; // disallow year arrows
         const r_date = epochToMoment(right_pane_date).startOf('month');
         if (diffInMonths(toMoment().startOf('month'), r_date) === 0) return true; // future months are disallowed
         return diffInMonths(r_date, date) !== 1;
     };
 
-    const jumpToCurrentMonth = () => {
+    /**
+     * Validate values to be date_from < date_to
+     *
+     * @param {moment.Moment} date
+     * @return {boolean}
+     */
+    const shouldDisableDate = (date: moment.Moment): boolean => {
+        return isPeriodDisabled(date.unix());
+    };
+
+    const jumpToCurrentMonth = (): void => {
         const current_month = toMoment().endOf('month').unix();
         setLeftPaneDate(epochToMoment(current_month).endOf('month').subtract(1, 'month').unix());
         setRightPaneDate(current_month);
     };
 
-    const navigateTo = e => {
-        setLeftPaneDate(subMonths(e, 1).unix());
-        setRightPaneDate(toMoment(e).unix());
-    };
-
-    const updateSelectedDate = e => {
+    const updateSelectedDate = (e: React.MouseEvent<HTMLElement>): void => {
```

Comment:
Do we need to define the return type here? I guess we can remove void type :thinking:

Reply:
> Do we need to define the return type here? I guess we can remove void type 🤔

I understand TS creates return type automatically, but if after some time we will change something can we be still sure the types are constant and haven't changed? Just wanna ask about this :)

Reply:
If I understand your question clearly, the returntype is base on what we returns in the function, so if we are going to change the function maybe we want to change the returntype too :thinking: 
BTW we don't define returntype for our functions unless we need it

Reply:
> If I understand your question clearly, the returntype is base on what we returns in the function, so if we are going to change the function maybe we want to change the returntype too 🤔 BTW we don't define returntype for our functions unless we need it

Ok, thanks for your time, in this case I'll accept your suggestions